pkg/pipeline/pipeline_prod.go

```diff
@@ -90,20 +116,80 @@ func newPipeline(input *InputBin, output *OutputBin) (*Pipeline, error) {
 	return &Pipeline{
 		pipeline: pipeline,
 		output:   output,
+		input:    input,
 		removed:  make(map[string]bool),
 		started:  make(chan struct{}, 1),
 		closed:   make(chan struct{}),
 	}, nil
 }
 
 func (p *Pipeline) Start() error {
+	bucket := "bucket-name-here"
+	object := "filename-here.mp4"
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		fmt.Errorf("storage.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*60)
+	defer cancel()
+
+	// Upload an object with storage.Writer.
+	wc := client.Bucket(bucket).Object(object).NewWriter(ctx)
+	wc.ContentType = "video/mp4"
+	wc.ChunkSize = 0 // note retries are not supported for chunk size 0.
+
+	asink := p.output.appSink
+	queue := p.output.queue
+	asink.SyncStateWithParent()
+	asink.SetWaitOnEOS(false)
+	queue.Link(asink.Element)
+	p.input.mux.Link(queue)
+
+	asink.SetCallbacks(&app.SinkCallbacks{
+		EOSFunc: func(sink *app.Sink) {
```

Comment:
I think this function needs to reupload the metadata at the start of the file (unless we remove `faststart` from mp4mux). It shouldn't need to post an EOS to the pipeline